modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesMessageMetadata.java

```diff
@@ -427,4 +431,36 @@ public void addProperty(String key, Object value) {
     public Object getProperty(String key) {
         return propertyMap.get(key);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof AndesMessageMetadata)){
+            return false;
+        }
+        AndesMessageMetadata that = (AndesMessageMetadata) o;
+        return getMessageID() == that.getMessageID();
+    }
+
+    @Override
+    public int hashCode() {
+        return (int) (getMessageID() ^ (getMessageID() >>> 32));
+    }
+
+    /**
+     * True if the expiration time is defined for the message.
+     */
+    public boolean isExpirationDefined() {
+        return isExpirationSet;
+    }
+
+    /**
+     * Set whether the message expiration value is defined.
+     * @param expirationSet
+     */
+    public void setIsExpirationDefined(boolean expirationSet) {
```

Comment:
expirationSet - argument doesn't aling with method name ( change the java doc as well)